download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/cache/Index.java

```diff
@@ -0,0 +1,32 @@
+package com.googlecode.download.maven.plugin.internal.cache;
+
+/**
+ * Cache index.
+ * @author Paul Polishchuk
+ * @since 1.3.1
+ */
+interface Index {
+
+    /**
+     * Adds given path to the index using url parameter as a key.
+     * @param url index key
+     * @param path index value
+     */
+    void put(String url, String path);
```

Comment:
@mickaelistria In this particular case, I would tend to disagree. Passing the `File` will be confusing, as client could think that Index will do something with files themselves but in fact it doesn't!

I think I got the source of the confusion - `FileIndex` class name. I meant "Persistent Index Backed by File", so it's not an "Index of Files"